src/main/java/com/labsynch/labseer/chemclasses/bbchem/CmpdRegMoleculeBBChemImpl.java

```diff
@@ -23,8 +23,9 @@
 
 	private BBChemStructureService bbChemStructureService;
 
-	public CmpdRegMoleculeBBChemImpl(BBChemParentStructure molecule) {
+	public CmpdRegMoleculeBBChemImpl(BBChemParentStructure molecule, BBChemStructureService bbChemStructureService) {
 		this.molecule = molecule;
+		this.bbChemStructureService = bbChemStructureService;
```

Comment:
Why is this needed?

Reply:
I couldn't get the @autowired annotation to work on this class so anywhere I knew up a CmpdRegMoleculeBBChemImpl I need to pass in the service or it ends up as null.  When debugging this new code, I found that if the CmpdRegMoleculeBBChemImpl class was newed up from a string, then I could save a molecule as it through a null pointer when calling bbchemStructureService.  I can't recall the specific issue in this pull requested that prompted this fix but it was important...